snuba/clusters/cluster.py

```diff
@@ -148,13 +184,27 @@ def get_writer(
             table_name, self.__host, self.__http_port, encoder, options, chunk_size
         )
 
+    def get_nodes(self) -> Sequence[ClickhouseNode]:
+        if self.__single_node:
```

Comment:
Why not just storing the list of ClickhouseNode in the constructor and return that list? If we are in single_node mode we return one node only. It seems to me there is no need right now to store the __single_node field, so no need to add this to the state. 
If/when we will need it we will store it.

Reply:
Sure, I mean this function isn't being called anywhere right now so in a way it's not needed yet. Just putting in place some pieces that will be useful later. But I think it's more useful to store the single node since other things like table name would depend on that. It also avoids always making the unnecessary request to system.clusters in the constructor.